src/vis/vis.js

```diff
@@ -319,6 +319,15 @@ var Vis = cdb.core.View.extend({
       this.https = true;
     }
 
+    // this checks that the screen is retina to add @2x to the url
+    if (window.devicePixelRatio > 1) {
+      data.layers.forEach(function(layer){
+        if(layer.type === "tiled"){
```

Comment:
My proposal is having different URLs for *normal* and retina resolution, if we detect we are in a retina display and we have a retina resolution URL for the basemap then we use it, otherwise fallback to the *normal* URL.